plugins/ibm_qradar/icon_ibm_qradar/actions/update_offense/action.py

```diff
@@ -0,0 +1,93 @@
+import insightconnect_plugin_runtime
+import requests
+from insightconnect_plugin_runtime.exceptions import ClientException, PluginException
+
+from .schema import UpdateOffenseInput, UpdateOffenseOutput, Component, Input, Output
+
+from icon_ibm_qradar.util.constants.endpoints import UPDATE_OFFENSES_ENDPOINT
+from icon_ibm_qradar.util.constants.messages import (
+    EMPTY_OFFENSE_ID_FOUND,
+    CLOSING_REASON_ID_NOT_PROVIDED,
+    CLOSING_REASON_ID_PROVIDED_FOR_OTHER_STATUS,
+    CLOSING_REASON_ID_PROVIDED_IS_NOT_INTEGER,
+)
+from icon_ibm_qradar.util.url import URL
+from icon_ibm_qradar.util.utils import prepare_request_params, handle_response
+
+
+class UpdateOffense(insightconnect_plugin_runtime.Action):
+    def __init__(self):
+        super().__init__(
+            name="update_offense",
+            description=Component.DESCRIPTION,
+            input=UpdateOffenseInput(),
+            output=UpdateOffenseOutput(),
+        )
+        self.endpoint = UPDATE_OFFENSES_ENDPOINT
+
+    def run(self, params={}):
```

Comment:
Please break this function up for maintainability and good practice - it is too large